### PR TITLE
Product list

### PR DESCRIPTION
**Pull request by NguyenMinhDuc98**

### CODE REVIEW

File: src/components/HeaderRight/index.tsx
Comment (khai-pi) [inline src/components/HeaderRight/index.tsx:1]: should create a HeaderRight.tsx file and export in index.tsx

File: src/screens/Products/List/index.tsx
Comment (khai-pi) [inline src/screens/Products/List/index.tsx:1]: create file and export in index

File: src/screens/Products/components/FloatingBar/index.tsx
Comment (khai-pi) [inline src/screens/Products/components/FloatingBar/index.tsx:1]: create file and export in index

File: src/screens/Products/components/ProductRow/index.tsx
Comment (khai-pi) [inline src/screens/Products/components/ProductRow/index.tsx:1]: create file and export in index

File: src/screens/Products/List/index.tsx [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+      ) : (
+        <>
+          <FlatList
+            data={data?.pages.flatMap((item: any) => item?.products)}
+            renderItem={({ item }) => <ProductRow item={item} />}
+            keyExtractor={item => item.id}
+            onEndReached={loadMore}
+            onEndReachedThreshold={50}
+            ListFooterComponent={
+              isFetchingNextPage ? <ActivityIndicator /> : null
```
Comment (khai-pi) [inline src/screens/Products/List/index.tsx:72]:
```suggestion
              isFetchingNextPage && <ActivityIndicator />
```

File: src/components/HeaderRight/index.tsx [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+import Ionicons from 'react-native-vector-icons/Ionicons';
+import MaterialIcons from 'react-native-vector-icons/MaterialIcons';
+import AntDesign from 'react-native-vector-icons/AntDesign';
+
+const HeaderRight = () => {
+  return (
+    <View style={styles.container}>
+      <TouchableOpacity
+        onPress={() => {
+          console.log();
```
Comment (tungUAS) [inline src/components/HeaderRight/index.tsx:12]: no console.log

File: src/customHooks/useInfiniteScroll.tsx [hunk 3]
```diff
@@ -0,0 +1,31 @@
+import { QueryKey, useInfiniteQuery } from '@tanstack/react-query';
+
+interface IUseInfiniteScroll {
+  queryKey: QueryKey;
+  fetchPage: (pageParam: number) => Promise<unknown>;
```
Comment (tungUAS) [inline src/customHooks/useInfiniteScroll.tsx:5]: unknown is quite bad

File: src/customHooks/useInfiniteScroll.tsx [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+  fetchPage,
+  options,
+}: IUseInfiniteScroll) => {
+  const { fetchNextPage, hasNextPage, isFetchingNextPage, ...result } =
+    useInfiniteQuery({
+      queryKey,
+      queryFn: ({ pageParam = 1 }: { pageParam: number }) =>
+        fetchPage(pageParam),
+      ...options,
+      getNextPageParam: (lastPage: any) => lastPage.products.length > 0,
```
Comment (tungUAS) [inline src/customHooks/useInfiniteScroll.tsx:20]: any is even worse

**Pull request merged**